src/test/java/org/elasticsearch/indices/recovery/RecoveryTargetTests.java

```diff
@@ -0,0 +1,108 @@
+            String id = Integer.toString(j);
+            client().prepareIndex("test", "type1", id).setSource("text", "sometext").get();
+        }
+        client().admin().indices().prepareFlush("test").setWaitIfOngoing(true).setForce(true).get();
+        RecoveryTarget recoveryTarget = getInstanceFromNode(RecoveryTarget.class);
+        IndexService idxService = getInstanceFromNode(IndicesService.class).indexService("test");
+        Store store = idxService.shard(0).store();
+        store.incRef();
+        try {
+            DiscoveryNode discoveryNode = new DiscoveryNode("123", new LocalTransportAddress("123"), Version.CURRENT);
+            Map<String, StoreFileMetaData> metaDataMap = recoveryTarget.existingFiles(discoveryNode, store, randomBoolean());
+            assertTrue(metaDataMap.size() > 0);
+            int iters = randomIntBetween(10, 20);
+            for (int i = 0; i < iters; i++) {
+                Version version = randomVersion();
+                DiscoveryNode discoNode = new DiscoveryNode("123", new LocalTransportAddress("123"), version);
+                Map<String, StoreFileMetaData> map = recoveryTarget.existingFiles(discoNode, store, false);
+                if (version.before(Version.V_1_4_0)) {
+                    assertTrue(map.isEmpty());
+                } else {
+                    assertEquals(map.size(), metaDataMap.size());
+                }
+
+            }
+        } finally {
+            store.decRef();
+        }
+
+    }
+
+    /**
+     * we fail to recover from pre 1.3.2 if compression is enabled
+     */
+    public void testFailPre1_3_2_WithCompression() throws IOException {
+        createIndex("test");
+
+        int numDocs = scaledRandomIntBetween(10, 100);
+        for (int j = 0; j < numDocs; ++j) {
+            String id = Integer.toString(j);
+            client().prepareIndex("test", "type1", id).setSource("text", "sometext").get();
+        }
+        client().admin().indices().prepareFlush("test").setWaitIfOngoing(true).setForce(true).get();
+        RecoveryTarget recoveryTarget = getInstanceFromNode(RecoveryTarget.class);
+        IndexService idxService = getInstanceFromNode(IndicesService.class).indexService("test");
+        Store store = idxService.shard(0).store();
+        store.incRef();
+        try {
+            int iters = randomIntBetween(10, 20);
+            for (int i = 0; i < iters; i++) {
+                Version version = randomVersion();
+                DiscoveryNode discoNode = new DiscoveryNode("123", new LocalTransportAddress("123"), version);
+                boolean withCompression = randomBoolean();
+                try {
+                    recoveryTarget.existingFiles(discoNode, store, withCompression);
+                    assertTrue(discoNode.version() + "  " + withCompression, version.onOrAfter(Version.V_1_3_2) || withCompression == false);
+                } catch (ElasticsearchIllegalStateException ex) {
+                    // all is good
```

Comment:
can we check that we expect this exception? i.e., when version is before 1.3.2 and compression is on?